tortoise/queryset.py

```diff
@@ -1596,13 +1608,23 @@ def _make_query(self) -> None:
             self.query._use_indexes = []
             self.query = self.query.use_index(*self.use_indexes)
 
-    def __await__(self) -> Generator[Any, None, Union[List[dict], Dict]]:
+    @overload
+    def __await__(
+        self: "ValuesQuery[Literal[False]]",
+    ) -> Generator[Any, None, List[Dict[str, Any]]]:
+        ...
+
+    @overload
+    def __await__(self: "ValuesQuery[Literal[True]]") -> Generator[Any, None, Dict[str, Any]]:
+        ...
+
+    def __await__(self) -> Generator[Any, None, Union[List[Dict[str, Any]], Dict[str, Any]]]:
         if self._db is None:
             self._db = self._choose_db()  # type: ignore
         self._make_query()
         return self._execute().__await__()  # pylint: disable=E1101
 
-    async def __aiter__(self) -> AsyncIterator[dict]:
+    async def __aiter__(self: "ValuesQuery[Any]") -> AsyncIterator[dict[str, Any]]:
```

Comment:
Seems like the following makes a bit more sense with regards to typehints and logic because it doesn't make much sense to use `__aiter__` when SINGLE is True because we would be iterating over keys of a dict. 

```suggestion
    async def __aiter__(self: "ValuesQuery[Literal[False]]") -> AsyncIterator[dict[str, Any]]:
```